layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -660,7 +660,9 @@ buffer in a split window."
       ('frame (select-frame (make-frame))))
     ;; Prompt to save on `save-some-buffers' with positive PRED
     (with-current-buffer newbuf
-      (setq-local buffer-offer-save t))
+      (progn
```

Comment:
The `progn` isn't needed here, `with-current-buffer` runs all functions in it's body.

One place where a `progn` is needed is in an if statement:
```elisp
(if this-is-non-nil
    (run-this)
  (otherwise-run-this))
```
that works as expected, if `this-is-non-nil` is non-nil, then `(run-this)` runs, otherwise `(otherwise-run-this)` runs.
```elisp
(if this-is-non-nil
    (run-this)
  (and-run-this)
  (otherwise-run-this))
```
Here only `(run-this)` runs if `this-is-non-nil` is non-nil.

The `if` functions docstring says:
>THEN must be one expression, but ELSE... can be zero or more expressions.

Therefore if `this-is-non-nil` is `nil` then both `(and-run-this)` and `(otherwise-run-this)` runs.

To run the first two commands when `this-is-non-nil` is non-nil, we need a `progn`
```elisp
(if this-is-non-nil
    (progn
      (run-this)
      (and-run-this))
  (otherwise-run-this))
```

Reply:
Ok, it seems I missed the `&rest` in the documentation of `with-current-buffer`... `<f1> f with-current-buffer RET` said `body`, not `body...`, so at a quick glance I thought it needs a `progn`. :)